tests/service/XmlDocTests.fs

```diff
@@ -10,10 +10,119 @@ module Tests.Service.XmlDocTests.XmlDoc
 open FSharp.Compiler.CodeAnalysis
 open FSharp.Compiler.Service.Tests.Common
 open FSharp.Compiler.Symbols
+open FSharp.Compiler.Syntax
 open FSharp.Test.Compiler
 open FsUnit
 open NUnit.Framework
-open FSharp.Compiler.Diagnostics
+
+let (|Types|TypeSigs|) = function
```

Comment:
@DedSec256 I think it would probably be good to have these active patterns extracted into some common place. There's nothing xmldoc-specific in them, and it could be great to have util code to reuse it in other parser-related tests.